nixops/util.py

```diff
@@ -149,21 +150,33 @@ def _transform_value(key: Any, value: Any) -> Any:
             if not ann:
                 return value
 
-            if inspect.isclass(ann) and issubclass(ann, ImmutableValidatedObject):
+            if isclass(ann) and issubclass(ann, ImmutableValidatedObject):
                 value = ann(**value)
 
-            # Support Sequence[ImmutableValidatedObject]
-            if isinstance(value, tuple) and not isinstance(ann, str):
-                new_value = []
-                for v in value:
-                    for subann in ann.__args__:  # type: ignore
-                        if inspect.isclass(subann) and issubclass(
-                            subann, ImmutableValidatedObject
-                        ):
-                            new_value.append(subann(**v))
-                        else:
-                            new_value.append(v)
-                value = tuple(new_value)
+            # Support containers of ImmutableValidatedObjects
+            match typing.get_origin(ann):
+
+                case collections.abc.Sequence:
+                    new_value: Any = []
```

Comment:
```suggestion
                    new_value: List[Any] = []
```

or maybe this could be more specific.

If the mypy-ratchet job still fails after that, I won't complain.

Reply:
Did this and added `Dict` to the mapping one + renamed them to avoid variable redefinition. The ratchet still fails locally for me, though. Maybe it's time to disable it?

Reply:
> Maybe it's time to disable it?

I think we should keep it around, so that regressions in type coverage don't go unnoticed, but also it's ok to override it when we don't seem to have a good way to improve the types in the change.

Reply:
Yeah, it's just not very helpful, since it doesn't actually point to where you have undeclared types. Does it even do what it's supposed to? I really can't tell. It doesn't exactly make it easier to contribute to NixOps, though.